packages/http-client/src/proxy.ts

```diff
@@ -1,23 +1,23 @@
 export function getProxyUrl(reqUrl: URL): URL | undefined {
   let usingSsl = reqUrl.protocol === 'https:'
 
-  let proxyUrl: URL
   if (checkBypass(reqUrl)) {
-    return proxyUrl
+    return undefined
   }
 
-  let proxyVar: string
-  if (usingSsl) {
-    proxyVar = process.env['https_proxy'] || process.env['HTTPS_PROXY']
-  } else {
-    proxyVar = process.env['http_proxy'] || process.env['HTTP_PROXY']
-  }
+  const proxyVar = (() => {
```

Comment:
Why turn this into a function, shouldn't we set the string? No need to invoke this multiple times